client/lib/track-form/index.jsx

```diff
@@ -22,9 +22,10 @@ export const trackForm = WrappedComponent => class TrackFormComponent extends Co
 		this.setState( newState, callback );
 	};
 
-	replaceFields = ( fields, callback ) => {
+	replaceFields = ( fields, callback, keepOldFields = true ) => {
+		const oldFields = keepOldFields ? this.state.fields : {};
```

Comment:
Would `prevFields` feel more "React-y"? It's the nomenclature used in a lot of `component*` methods.